semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -761,20 +767,21 @@ let lval_of_ent env ent =
       | [] -> raise Impossible
       | x :: _ -> fresh_lval env x)
 
-(* just to ensure the code after does not call expr directly *)
-let expr_orig = expr
+(* just to ensure the code after does not call expr directly
+ * TODO: Remove this if we can't figure out why it was done in the first place... *)
+and expr_orig env ?void e = expr env ?void e
 
-let expr () = ()
+and dummy_expr () = ()
 
-let expr_with_pre_stmts env ?void e =
-  ignore (expr ());
+and expr_with_pre_stmts env ?void e =
```

Comment:
Why we suddenly need to make all those functions mutually recursive?

Reply:
Before we didn't account for `StmtExpr` so `expr` did not depend on `stmt`. But now `expr` and `stmt` are mutually recursive, so all those utility functions that are used by `stmt_aux` need to be part of the same `let-rec`.